mx.py

```diff
@@ -7053,6 +7053,43 @@ def run_java(args, nonZeroIsFatal=True, out=None, err=None, cwd=None, timeout=No
         jdk = get_jdk()
     return jdk.run_java(args, nonZeroIsFatal=nonZeroIsFatal, out=out, err=err, cwd=cwd, timeout=timeout, env=env, addDefaultArgs=addDefaultArgs)
 
+def run_java_min_heap(args, benchName='# MinHeap:', overheadFactor=1.5, referenceHeap=4096, out=None, err=None, cwd=None, timeout=None, env=None, addDefaultArgs=True, jdk=None):
+    """computes the minimum heap size required to run a Java program within a certain overhead factor"""
+
+    def run_with_heap(heap, args, timeout=timeout, suppressStderr=True, nonZeroIsFatal=False):
+        log('Trying with ' + str(heap) + 'MB of heap...')
+        with open(os.devnull, 'w') as fnull:
+            exitCode = run_java(['-Xmx' + str(heap) + 'M'] + args, nonZeroIsFatal=nonZeroIsFatal, out=out, err=fnull if suppressStderr else err, cwd=cwd, timeout=timeout, env=env, addDefaultArgs=addDefaultArgs)
```

Comment:
I think the last `-Xmx` arg is taken into account, maybe you could try to insert as the last JVM arg? I think only the `-cp` and `--classpath` JVM arg consume the next argument.